### PR TITLE
Improve error message for duplicate property with computed name

### LINKED ISSUE #27914: Improve error message for duplicate property with computed name
**TypeScript Version:**  3.2.0-dev.20181011

**Search Terms:**

**Code**

```ts
const key = 'bar';
const obj = {bar: 1, [key]: 2}
```

**Expected behavior:**

`Duplicate property 'bar'.` on `[key]` with a related span pointing to property `bar`.

**Actual behavior:**

`Duplicate declaration '[key]'.` on both property names.

**Playground Link:** http://www.typescriptlang.org/play/#src=const%20key%20%3D%20'bar'%3B%0D%0Aconst%20obj%20%3D%20%7Bbar%3A%201%2C%20%5Bkey%5D%3A%202%7D

**Related Issues:**

### PR DESCRIPTION
**Pull request by collin5**
Fixes #27914

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -6288,9 +6288,9 @@ namespace ts {
                         // If we have an existing early-bound member, combine its declarations so that we can
                         // report an error at each declaration.
                         const declarations = earlySymbol ? concatenate(earlySymbol.declarations, lateSymbol.declarations) : lateSymbol.declarations;
-                        const name = declarationNameToString(decl.name);
-                        forEach(declarations, declaration => error(getNameOfDeclaration(declaration) || declaration, Diagnostics.Duplicate_declaration_0, name));
-                        error(decl.name || decl, Diagnostics.Duplicate_declaration_0, name);
+                        const name = (<LiteralType>type).value || declarationNameToString(decl.name);
+                        forEach(declarations, declaration => error(getNameOfDeclaration(declaration) || declaration, Diagnostics.Duplicate_property_0, name));
```
Comment (ajafff) [inline src/compiler/checker.ts]: IMO you can remove this line. The correct error will be set in the next line anyway.
You could refactor this line to generate a related span for the error below that says `Property 'x' was also declared here.`.

**Comment by collin5** [comment]
@ajafff Thanks for the feedback. I've made the suggested changes 👍

**Comment by RyanCavanaugh** [comment]
Thanks and sorry for the delay!

**Comment by collin5** [comment]
No worries! Thank you 👍

**Pull request merged**